.github/workflows/pdm-lock-automation.yaml

```diff
@@ -0,0 +1,95 @@
+name: Automate pdm.lock
+
+on: push
```

Comment:
Remove this on push and add trigger for now. Once merged let's test this and make other changes if need. Then we can make this on pull_request.

Reply:
@ritwik-g done